app/src/main/java/com/nervousfish/nervousfish/activities/ContactReceivedHelper.java

```diff
@@ -41,17 +45,22 @@
      * @param contact  The newly received {@link Contact}
      */
     static void newContactReceived(final IDatabase database, final Activity activity, final Contact contact) {
+        if (contact == null) {
```

Comment:
I do not agree with that. The only way that contact can be null, is when your partner send you instead of his own Profile a null. That should not be possible, otherwise the time-consuming public key exchange is pretty useless.